Form/TypeSolver.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace FSi\Bundle\AdminSecurityBundle\Form;
+
+use Symfony\Component\Form\FormTypeInterface;
+use Symfony\Component\HttpKernel\Kernel;
+
+/**
+ * @internal
+ */
+final class TypeSolver
+{
+    /**
+     * Return FQCN form type or old style form type
+     *
+     * @param string $fqcnType
+     * @param string|FormTypeInterface $shortType
+     * @return string|FormTypeInterface
+     */
+    public static function getFormType($fqcnType, $shortType)
+    {
+        return self::isSymfony3FormNamingConvention() ? $fqcnType : $shortType;
+    }
+
+    /**
+     * @return bool
+     */
+    public static function isChoicesAsValuesOptionTrueByDefault()
+    {
+        return method_exists('Symfony\Component\Form\FormTypeInterface', 'configureOptions');
+    }
+
+    /**
+     * @return bool
+     */
+    public static function isSymfony3FormNamingConvention()
+    {
+        return Kernel::MAJOR_VERSION === 3;
```

Comment:
It would be better to check if `symfony/form` is in version 3.0 or later:
```
return !method_exist('\Symfony\Component\Form\FormTypeInterface', 'getName`);
```